awx/ui/client/src/management-jobs/card/card.controller.js

```diff
@@ -48,66 +48,84 @@ export default
                 $scope.submitJob = function (id, name) {
                     Wait('start');
                         defaultUrl = GetBasePath('system_job_templates')+id+'/launch/';
-                        CreateDialog({
-                            id: 'prompt-for-days'    ,
-                            title: name,
-                            scope: $scope,
-                            width: 500,
-                            height: 300,
-                            minWidth: 200,
-                            callback: 'PromptForDays',
-                            resizable: false,
-                            onOpen: function(){
-                                $scope.$watch('prompt_for_days_form.$invalid', function(invalid) {
-                                    if (invalid === true) {
-                                        $('#prompt-for-days-launch').prop("disabled", true);
-                                    } else {
-                                        $('#prompt-for-days-launch').prop("disabled", false);
-                                    }
-                                });
+                        var noModalJobs = ['Automation Insights Collection', 'Cleanup Expired Sessions', 'Cleanup Expired OAuth 2 Tokens']
+                        if(noModalJobs.includes(name)) {
+                                      var data = {};
+                                      Rest.setUrl(defaultUrl);
```

Comment:
I would like to see a new function added to this module called `launchManagementJob`. It should contain the launch code being duplicated [here](https://github.com/ansible/awx/blob/651019445b290745fc7f799d55de50f1e96895d1/awx/ui/client/src/management-jobs/card/card.controller.js#L54-L65) and [here](https://github.com/ansible/awx/blob/651019445b290745fc7f799d55de50f1e96895d1/awx/ui/client/src/management-jobs/card/card.controller.js#L108-L121). You can then call that function from both places.